File: LearnerBee/src/screens/InstructorListScreen.js
// src/screens/InstructorListScreen.js
import React, { useState, useEffect } from 'react';
import { 
  View, 
  Text, 
  StyleSheet, 
  FlatList, 
  TouchableOpacity, 
  ScrollView,
  SafeAreaView
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import { getInstructors, locations } from '../services/mockData';
import InstructorCard from '../components/InstructorCard';
import FilterButton from '../components/FilterButton';

const InstructorListScreen = ({ navigation, route }) => {
  const [instructors, setInstructors] = useState([]);
  const [selectedLocation, setSelectedLocation] = useState('All Locations');
  const [filters, setFilters] = useState({});
  const [instructorCount, setInstructorCount] = useState(0);

  useEffect(() => {
    // Get filters from route params if they exist
    const routeFilters = route.params?.filters || {};
    setFilters(routeFilters);
    
    // Apply filters
    const filteredInstructors = getInstructors({
      ...routeFilters,
      location: selectedLocation !== 'All Locations' ? selectedLocation : undefined
    });
    setInstructors(filteredInstructors);
    setInstructorCount(filteredInstructors.length);
  }, [route.params, selectedLocation]);

  const renderLocationButton = (location) => {
    return (
      <FilterButton
        key={location}
        title={location}
        isActive={selectedLocation === location}
        onPress={() => setSelectedLocation(location)}
      />
    );
  };

  return (
    <SafeAreaView style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.instructorCount}>{instructorCount} instructors</Text>
      </View>
      
      <View style={styles.filterBar}>
        <ScrollView horizontal showsHorizontalScrollIndicator={false} contentContainerStyle={styles.filterScroll}>
          <FilterButton
            title="Price"
            onPress={() => navigation.navigate('Filter', { activeTab: 'price', currentFilters: filters })}
          />
          <FilterButton
            title="Rating"
            onPress={() => navigation.navigate('Filter', { activeTab: 'rating', currentFilters: filters })}
          />
          <FilterButton
            title="Location"
            isActive={true}
          />
          <TouchableOpacity
            style={styles.sortButton}
            onPress={() => navigation.navigate('Filter', { activeTab: 'sort', currentFilters: filters })}
          >
            <Icon name="sort" size={16} color="#333" />
            <Text style={styles.sortText}>Sort</Text>
          </TouchableOpacity>
        </ScrollView>
      </View>
      
      <ScrollView 
        horizontal 
        showsHorizontalScrollIndicator={false}
        contentContainerStyle={styles.locationScroll}
      >
        {renderLocationButton('All Locations')}
        {locations.map(location => renderLocationButton(location))}
      </ScrollView>
      
      <FlatList
        data={instructors}
        keyExtractor={(item) => item.id}
        renderItem={({ item }) => <InstructorCard instructor={item} />}
        contentContainerStyle={styles.listContent}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#F8F8F8',
  },
  header: {
    paddingHorizontal: 16,
    paddingVertical: 12,
    borderBottomWidth: 1,
    borderBottomColor: '#EEEEEE',
  },
  instructorCount: {
    fontSize: 24,
    fontWeight: 'bold',
  },
  filterBar: {
    paddingVertical: 12,
    borderBottomWidth: 1,
    borderBottomColor: '#EEEEEE',
  },
  filterScroll: {
    paddingHorizontal: 16,
  },
  locationScroll: {
    paddingVertical: 12,
    paddingHorizontal: 16,
  },
  sortButton: {
    flexDirection: 'row',
    alignItems: 'center',
    paddingHorizontal: 16,
    paddingVertical: 8,
    borderRadius: 20,
    borderWidth: 1,
    borderColor: '#E0E0E0',
    backgroundColor: 'white',
  },
  sortText: {
    marginLeft: 4,
    fontSize: 14,
    color: '#333',
  },
  listContent: {
    paddingVertical: 8,
  },
});

export default InstructorListScreen;

File: LearnerBee/lib/screens/instructor_list_screen.dart
 import 'package:flutter/material.dart';
import '../models/instructor.dart';
import '../services/instructor_service.dart';
import '../widgets/instructor_card.dart';
import 'filter_screen.dart';

class InstructorListScreen extends StatefulWidget {
  @override
  _InstructorListScreenState createState() => _InstructorListScreenState();
}

class _InstructorListScreenState extends State<InstructorListScreen> {
  List<Instructor> instructors = [];
  String selectedLocation = 'All Locations';
  Map<String, dynamic> filters = {};
  
  @override
  void initState() {
    super.initState();
    _loadInstructors();
  }
  
  void _loadInstructors() {
    setState(() {
      instructors = InstructorService.getInstructors(
        location: selectedLocation != 'All Locations' ? selectedLocation : null,
        minPrice: filters['minPrice'],
        maxPrice: filters['maxPrice'],
        minRating: filters['minRating'],
        sortBy: filters['sortBy'],
      );
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Find Instructors'),
      ),
      body: Column(
        children: [
          // Instructor count header
          Padding(
            padding: EdgeInsets.all(16),
            child: Align(
              alignment: Alignment.centerLeft,
              child: Text(
                '${instructors.length} instructors',
                style: TextStyle(
                  fontSize: 24,
                  fontWeight: FontWeight.bold,
                ),
              ),
            ),
          ),
          
          // Filter buttons
          Container(
            padding: EdgeInsets.symmetric(vertical: 12),
            decoration: BoxDecoration(
              border: Border(
                bottom: BorderSide(color: Colors.grey[300]!),
              ),
            ),
            child: SingleChildScrollView(
              scrollDirection: Axis.horizontal,
              padding: EdgeInsets.symmetric(horizontal: 16),
              child: Row(
                children: [
                  _buildFilterButton('Price', onPressed: () async {
                    final result = await Navigator.push(
                      context,
                      MaterialPageRoute(
                        builder: (context) => FilterScreen(
                          activeTab: 'price',
                          currentFilters: filters,
                        ),
                      ),
                    );
                    
                    if (result != null) {
                      setState(() {
                        filters = result;
                        _loadInstructors();
                      });
                    }
                  }),
                  SizedBox(width: 8),
                  _buildFilterButton('Rating', onPressed: () async {
                    final result = await Navigator.push(
                      context,
                      MaterialPageRoute(
                        builder: (context) => FilterScreen(
                          activeTab: 'rating',
                          currentFilters: filters,
                        ),
                      ),
                    );
                    
                    if (result != null) {
                      setState(() {
                        filters = result;
                        _loadInstructors();
                      });
                    }
                  }),
                  SizedBox(width: 8),
                  _buildFilterButton('Location', isActive: true),
                  SizedBox(width: 8),
                  _buildFilterButton('Sort', icon: Icons.sort, onPressed: () async {
                    final result = await Navigator.push(
                      context,
                      MaterialPageRoute(
                        builder: (context) => FilterScreen(
                          activeTab: 'sort',
                          currentFilters: filters,
                        ),
                      ),
                    );
                    
                    if (result != null) {
                      setState(() {
                        filters = result;
                        _loadInstructors();
                      });
                    }
                  }),
                ],
              ),
            ),
          ),
          
          // Location filter
          Container(
            height: 50,
            padding: EdgeInsets.symmetric(vertical: 8),
            child: ListView(
              scrollDirection: Axis.horizontal,
              padding: EdgeInsets.symmetric(horizontal: 16),
              children: InstructorService.locations.map((location) {
                return Padding(
                  padding: EdgeInsets.only(right: 8),
                  child: _buildLocationButton(location),
                );
              }).toList(),
            ),
          ),
          
          // Instructor list
          Expanded(
            child: ListView.builder(
              padding: EdgeInsets.symmetric(vertical: 8),
              itemCount: instructors.length,
              itemBuilder: (context, index) {
                return InstructorCard(instructor: instructors[index]);
              },
            ),
          ),
        ],
      ),
    );
  }

  Widget _buildFilterButton(String title, {
    IconData? icon, 
    bool isActive = false, 
    VoidCallback? onPressed,
  }) {
    return ElevatedButton(
      onPressed: onPressed,
      style: ElevatedButton.styleFrom(
        foregroundColor: isActive ? Colors.white : Colors.grey[700],
        backgroundColor: isActive ? Theme.of(context).primaryColor : Colors.white,
        shape: RoundedRectangleBorder(
          borderRadius: BorderRadius.circular(20),
          side: BorderSide(color: isActive ? Colors.transparent : Colors.grey[300]!),
        ),
        elevation: 0,
        padding: EdgeInsets.symmetric(horizontal: 16, vertical: 8),
      ),
      child: Row(
        children: [
          if (icon != null) ...[
            Icon(icon, size: 16),
            SizedBox(width: 4),
          ],
          Text(title),
        ],
      ),
    );
  }

  Widget _buildLocationButton(String location) {
    final isSelected = selectedLocation == location;
    
    return ElevatedButton(
      onPressed: () {
        setState(() {
          selectedLocation = location;
          _loadInstructors();
        });
      },
      style: ElevatedButton.styleFrom(
        foregroundColor: isSelected ? Colors.white : Colors.grey[700],
        backgroundColor: isSelected ? Theme.of(context).primaryColor : Colors.white,
        shape: RoundedRectangleBorder(
          borderRadius: BorderRadius.circular(20),
          side: BorderSide(color: isSelected ? Colors.transparent : Colors.grey[300]!),
        ),
        elevation: 0,
        padding: EdgeInsets.symmetric(horizontal: 16, vertical: 8),
      ),
      child: Text(location),
    );
  }
}